### ISSUE TITLE
algorithmOutput object stacks the correct answers each time the algorithm runs 

### ISSUE THREAD

**Issue by szhabolcs** [ToastedWaffle/graph-theory-visualization#1]
Resetting the graph animation should reset the output object, but it doesn't do it. This leads to other issues where changing the graph doesn't change the animation.
![image](https://user-images.githubusercontent.com/54114237/85166462-ffbe5880-b26f-11ea-9854-b4ef5687674e.png)
As you can see the graph shown adds exactly 4 more moves to the existing output.
![image](https://user-images.githubusercontent.com/54114237/85166576-28465280-b270-11ea-8697-e467f71565a3.png)

**Comment by szhabolcs** [comment]
Upon further investigating, I found out that the object does get cleared, but because of an unknown reason, after running an animation, and then resetting it, leads to adding more and more unnecessary steps to the object.
Example:
We start out with this graph:
![image](https://user-images.githubusercontent.com/54114237/85418871-06074a00-b57a-11ea-8b57-dae73ce594e7.png)
running the algorithm for the first time works flawlessly, and the object has the correct steps:
![image](https://user-images.githubusercontent.com/54114237/85418998-3353f800-b57a-11ea-9a79-75f0bccc785a.png)
pressing the reset button, and triggering the animation again adds the correct steps **twice**:
![image](https://user-images.githubusercontent.com/54114237/85419292-92b20800-b57a-11ea-9f9a-68da551eb7b8.png)
doing so once more, has the same result; it adds 5 more correct steps:
![image](https://user-images.githubusercontent.com/54114237/85419409-bbd29880-b57a-11ea-8a3a-3c2a94a75198.png)

So in conclusion, this issue needs some more digging, but we know for now, that resetting seems to cause the problem.

**EDIT:**
I've tried the other undirected graph algorithms and they all cause this effect of adding the steps multiple times

**Comment by szhabolcs** [comment]
Finally found the issue! 😄

After spending some time using the Chrome debugger I figured out, that the issue in in fact related to jQuery, because this function
![image](https://user-images.githubusercontent.com/54114237/86265497-82c6a380-bbcc-11ea-9a18-067372f155d9.png)
is fired more than once, in fact it is fired **one more time**, each time.

adding a console log, to one of the algorithms reveals that the algorithm function is called multiple times, and this is why the answers get stacked in the algorithmOutput object
![image](https://user-images.githubusercontent.com/54114237/86265992-3cbe0f80-bbcd-11ea-881e-d5564d1ce513.png)

running for the first time
![image](https://user-images.githubusercontent.com/54114237/86267470-6e37da80-bbcf-11ea-8bb5-d8bfd502f557.png)
running it the second time, shows that the algorithm has been called 2 times instead of 1 (plus the 1 before)
![image](https://user-images.githubusercontent.com/54114237/86267519-7e4fba00-bbcf-11ea-8f98-9c714fd024fc.png)

I think the issue has to do with delegation, because the nodeSelect function is called using delegation
![image](https://user-images.githubusercontent.com/54114237/86267705-cff84480-bbcf-11ea-896d-cf54dea9addb.png)
